src/Disciplina/DB/Class.hs

```diff
@@ -0,0 +1,12 @@
+module Disciplina.DB.Class where
+
+import Universum
+
+-- | Read-only interface to key-value DB storing serialized data.
+class Monad m => MonadDBRead m where
+    dbGet :: ByteString -> m (Maybe ByteString)
+
+-- | Full interface to key-value DB.
+class MonadDBRead m => MonadDB m where
+    dbPut :: ByteString -> ByteString -> m ()
+    dbDelete :: ByteString -> m ()
```

Comment:
What will be stored there?

I have `auth-data-structures:Data.Tree.AVL.KVStoreMonad..KVStoreMonad` class.

How will it interact with your class?

Reply:
>How will it interact with your class?
Obviously:
```
retrieve = fmap decodeStrict . dbGet . encodeStrict
store k = dbPut k . encodeStrict
```